src/misc.cpp

```
@@ -221,6 +225,19 @@ const std::string compiler_info() {
   return compiler;
 }
 
+/// large_page_alloc() tries to allocate aligned large page memory on systems
+/// where available, if no large page memory is available, small page memory
+/// is allocated instead.
+void* large_page_alloc(size_t alignment, size_t size) {
+  auto addr = aligned_alloc(alignment, size);
+
+#ifdef __linux__
+  if (addr)
+      madvise(addr, size, MADV_HUGEPAGE);
```

This will silently fail if addr is not page-aligned.